datafusion/sqllogictest/test_files/case.slt

```diff
@@ -289,12 +289,22 @@ query B
 select case when a=1 then false end from foo;
 ----
 false
-false
-false
-false
-false
-false
+NULL
+NULL
+NULL
+NULL
+NULL
 
+query IBB
```

Comment:
Here is the postgres results (same as this PR)

```sql
postgres=# SELECT c,
    CASE WHEN c > 0 THEN true END AS c1,
    CASE WHEN c > 0 THEN true ELSE false END AS c2
FROM (VALUES (1), (0), (-1), (NULL)) AS t(c);
 c  | c1 | c2
----+----+----
  1 | t  | t
  0 |    | f
 -1 |    | f
    |    | f
(4 rows)
```

Reply:
Thanks for double checking me!
I also posted results from PostgreSQL (and Trino) in the issue https://github.com/apache/datafusion/issues/14343